packages/nox/test/tpetra/DS6.5.1/DennisSchnabelTpetra.hpp

```diff
@@ -0,0 +1,127 @@
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Roger Pawlowski (rppawlo@sandia.gov) or
+// Eric Phipps (etphipp@sandia.gov), Sandia National Laboratories.
+// ************************************************************************
+//  CVS Information
+//  $Source$
+//  $Author$
+//  $Date$
+//  $Revision$
+// ************************************************************************
+//@HEADER
+
+// Finite Element Problem Class
+/* Provides function (RHS) and Jacobian evaluations for the
+ * 2 equation test for quadratic and cubic line searches
+ * from Dennis & Schnabel's book, chp 6. example 6.5.1
+ *
+ *    U0**2 + U1**2 - 2 = 0
+ *    exp(U0-1) + U1**3 -2 = 0
+ */
+
+#ifndef _NOX_EXAMPLE_DENNIS_SCHNABEL_651_H
+#define _NOX_EXAMPLE_DENNIS_SCHNABEL_651_H
+
+// #include "NOX_Epetra_Interface_Required.H"
+#include "Teuchos_RCP.hpp"
+#include "NOX_TpetraTypedefs.hpp"
+
+// Forward Declarations
+// class Tpetra_Comm;
+// class Tpetra_Map;
+// class Tpetra_Vector;
+// class Tpetra_Import;
+// class Tpetra_CrsGraph;
+// class Tpetra_CrsMatrix;
+
+// Flag to tell the evaluate routine what objects to fill
+enum FillType {F_ONLY, MATRIX_ONLY, ALL};
+
+// Finite Element Problem Class
+class DennisSchnabel {
+
+public:
+
+  // Constructor
+  DennisSchnabel(int NumGlobalElements, Teuchos::RCP< const Teuchos::Comm<int> > &Comm);
+
+  // Destructor
+  ~DennisSchnabel();
+
+  // Evaluates the function (RHS) and/or the Jacobian using the solution
+  // values in solnVector.
+  bool evaluate( Tpetra::CombineMode fillType, // CWS: check
```

Comment:
Be carefull. The previous `NOX::Epetra::Interface::Required::FillType` type is [this](https://docs.trilinos.org/dev/packages/nox/doc/html/classNOX_1_1Epetra_1_1Interface_1_1Required.html). And `Tpetra::CombineMode` [here](https://docs.trilinos.org/dev/packages/nox/doc/html/classNOX_1_1Epetra_1_1Interface_1_1Required.html) is something else, but you need it in `.cpp` file [here](https://github.com/NexGenAnalytics/Trilinos/pull/193/files#diff-07d339264728d2fe8ae8b3bafa182857a7425a055c31e9aee64309c3c2c88095R142)

Reply:
I know your PR is wip but I'm pointing it out because I haven't found an equivalent for `NOX::Epetra::Interface::Required::FillType` and it's blocking me and the whole NOX package.

Reply:
Thanks @antoinemeyer5 !